### PR TITLE
GMUPDATETRANS saved in flow_trans_data.txt

### PR DESCRIPTION
**Pull request by jaewooan**
Commit title: "GMUPDATETRANS is saved in flow_trans_data_txt"
1. GMUPDATETRANS is printed in the function save_output
2. New variable members are defined in the class face: conType, ConCV, ConTr, ConCV, ConArea, ZVolumeFactor, ConPerm.
3. These members will be printed for GMUPDATETRANS.

### CODE REVIEW

**Comment by ishovkun** [comment]
Are you sure you want to merge it into master???

**Comment by jaewooan** [comment]
> Are you sure you want to merge it into master???

Sorry, I made a mistake. I intended to merge it to 'develop.'. :)

File: src/gprs-data/OutputDataGPRS.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ void OutputDataGPRS::saveGeometry(const std::string & output_path)
   // gprs output
 
   // GEOMETRY
-  const std::string outstring = output_path + data.config.mechanics_domain_file;
+  const std::string outstring = output_path + "gm_geometry.txt";// Jaewoo An data.config.mechanics_domain_file;
```
Comment (ishovkun) [inline src/gprs-data/OutputDataGPRS.cpp]: Avoid hard-coded paths. Things get messed up quickly.

File: src/gprs-data/FlowData.hpp [hunk 2]
```diff
@@ -23,6 +23,14 @@ struct FaceData
...
   double transmissibility;
   double thermal_conductivity;
+
+  std::size_t ConType;
+  std::size_t ConN;
+  std::vector<double> ConCV;
+  std::vector<double> ConTr;
+  std::vector<double> ConArea;
+  std::vector<double> ZVolumeFactor;
+  std::vector<double> ConPerm;
```
Comment (ishovkun) [inline src/gprs-data/FlowData.hpp]: Don't start non-class variables with a capital letters. Also, if a variable name isn't completely self-explanatory, add documenting comments.

File: src/gprs-data/FlowData.hpp [hunk 3]
```diff
@@ -23,6 +23,14 @@ struct FaceData
 {
   double transmissibility;
   double thermal_conductivity;
+
+  std::size_t ConType;
+  std::size_t ConN;
```
Comment (ishovkun) [inline src/gprs-data/FlowData.hpp]: You don't need ConN since you can get it as ConCV.size()

File: src/gprs-data/FlowData.cpp [hunk 4]
```diff
@@ -38,17 +38,17 @@ void FlowData::merge_elements(const std::size_t updated_element,
   // const double v0 = volumes[u];
   // const double v1 = volumes[d];
   // volumes[u] += v1;
-  // poro[u] += (v0*poro[u] + v1*poro[d]) / (v0 + v1);
-  // depth[u] += (v0*depth[u] + v1*depth[d]) / (v0 + v1);
+  // poro[u] = (v0*poro[u] + v1*poro[d]) / (v0 + v1);
+  // depth[u] = (v0*depth[u] + v1*depth[d]) / (v0 + v1);
 
   // for (std::size_t i=0; i<custom_data[u].size(); ++i)
   //   custom_data[u][i] = (v0*custom_data[u][i] + v1*custom_data[d][i]) / (v0 + v1);
```
Comment (ishovkun) [inline src/gprs-data/FlowData.cpp]: remove lines that were commented out

File: src/gprs-data/SimdataConfig.hpp [hunk 5]
```diff
@@ -100,7 +100,8 @@ struct SimdataConfig
   std::vector<BCNodeConfig>            bc_nodes;
   std::vector<WellConfig>              wells;
 
-  EDFMMethod edfm_method = EDFMMethod::simple;       // method to simulate flow in embedded fracs
+  EDFMMethod edfm_method = EDFMMethod::simple;
+  //EDFMMethod edfm_method = EDFMMethod::projection;       // method to simulate flow in embedded fracs; EDFMMethod::simple or EDFMMethod::projection
```
Comment (ishovkun) [inline src/gprs-data/SimdataConfig.hpp]: remove commented out line

File: src/gprs-data/transes.cpp [hunk 6]
```diff
@@ -1420,6 +1480,7 @@ void CalcTranses::save_output(const FlowData    & data,
 {
   const std::string fname_cell_data = "fl_cell_data.txt";
   const std::string fname_face_data = "fl_face_data.txt";
+  const std::string fname_gmupdate_data = "fl_trans_data.txt"; // 191217 Jaewoo An
```
Comment (ishovkun) [inline src/gprs-data/transes.cpp]: just save it in fname_face_data.

File: src/msh2gprs/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+find_package(Qt5 COMPONENTS Widgets REQUIRED)
+
+add_executable(msh2gprs
+  main.cpp
+  mainwindow.cpp
+  mainwindow.h
+  mainwindow.ui
+)
+
+target_link_libraries(msh2gprs PRIVATE Qt5::Widgets)
```
Comment (ishovkun) [inline src/msh2gprs/CMakeLists.txt]: Why do you need QT???

**Review by ishovkun — CHANGES_REQUESTED** [review]

File: src/msh2gprs/CMakeLists.txt (hunk 7 above)
Comment (jaewooan) [reply src/msh2gprs/CMakeLists.txt]: On Linux, I used QT to debug MSH2GPRS. It is better to remove QT parts.  I didn't recognize qt-related part is added to CMakeList. I will modify the lines 14 and 23 and pull request again.

**Comment by jaewooan** [comment]
I will send modified pull request

**Comment by ishovkun** [comment]
Looks good.

**Pull request merged**